### PR TITLE
STOP_PROCESSING now removes the datum from currentrun as well

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

title

## Why It's Good For The Game

STOP_PROCESSING should make sure anything that's supposed to stop processing, well, stops processing. This can potentially cause runtimes if, say, something is Destroyed() and shouldn't be processing and processes yet another cycle and the process() proc tries to access something that no longer exists.

## Changelog
:cl:
fix: STOP_PROCESSING now removes from currentrun to prevent another process cycle from happening where it shouldn't.
/:cl:

### CODE REVIEW

**Comment by kevinz000** [comment]
code/datums/weather/weather.dm:105:error: SSweather.currentrun: undefined var
HNNNNNG

**Comment by kevinz000** [comment]
arlight well sOMEONE decided to use it for a non processing subsystem so HERE I GO BREAKING SHIT AGAIN

**Comment by Cyberboss** [comment]
le mayo.

but good pr

**Comment by spookydonut** [comment]
Is this actually a good idea, given currentrun is for when a subsystem resumes after overrunning?

**Comment by AnturK** [comment]
Why wouldn't it be. The only issue with these is usually if it's cheaper to check for nulls/deletions or do this.

**Comment by ShizCalev** [comment]
doesn't compile.

File: code/controllers/subsystem/processing/weather.dm [hunk 1]
```diff
@@ -13,16 +13,8 @@ SUBSYSTEM_DEF(weather)
 	var/list/eligible_zlevels = list()
```
Comment (Dennok) [inline code/controllers/subsystem/processing/weather.dm]: Forgot to remove this on 10 `var/list/processing = list()`

**Pull request closed without merging**